### PR TITLE
Supports Report

### PR DESCRIPTION
**Pull request by hfazai**
Basic report syntax:
````Kotlin
  class SimpleReport : Report() {
    val field1 = field(StringTestType()) {
      label = "field1"
    }
    val field2 = field(LongTestType()) {
      label = "field2"
    }

    init {
      add {
        this[field1] = "test1"
        this[field2] = 64L
      }
      add {
        this[field1] = "test2"
        this[field2] = 32L
      }
    }
  }
````

### CODE REVIEW

File: galite-core/src/main/kotlin/org/kopi/galite/visual/addons/common/Window.kt [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ */
+
+package org.kopi.galite.visual.addons.common
+
+/**
+ * Abstract class for all window components.
+ */
+abstract class Window {
+
+}
```
Comment (ZMarouani) [inline galite-core/src/main/kotlin/org/kopi/galite/visual/addons/common/Window.kt:25]: Empty abstract class , should contain fields or methods to be extended from

File: galite-core/src/main/kotlin/org/kopi/galite/visual/addons/report/Table.kt [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+ */
+
+package org.kopi.galite.visual.addons.report
+
+/**
+ * Data table for of a report.
+ */
+class Table {
+
+}
```
Comment (ZMarouani) [inline galite-core/src/main/kotlin/org/kopi/galite/visual/addons/report/Table.kt:25]: Empty class has no utility

File: galite-core/src/main/kotlin/org/kopi/galite/visual/addons/report/VReport.kt [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+import org.kopi.galite.visual.addons.common.Window
+
+/**
+ * Visual class for a report.
+ */
+class VReport: Window() {
+  init {
+
+  }
```
Comment (ZMarouani) [inline galite-core/src/main/kotlin/org/kopi/galite/visual/addons/report/VReport.kt:28]: The constructor is empty

File: galite-core/src/main/kotlin/org/kopi/galite/visual/addons/common/Window.kt (hunk 1 above)
Comment (hfazai) [reply galite-core/src/main/kotlin/org/kopi/galite/visual/addons/common/Window.kt:25]: indead, it's to be implemeted later when adding visual report

File: galite-core/src/main/kotlin/org/kopi/galite/visual/addons/report/Table.kt (hunk 2 above)
Comment (hfazai) [reply galite-core/src/main/kotlin/org/kopi/galite/visual/addons/report/Table.kt:25]: Same

File: galite-core/src/main/kotlin/org/kopi/galite/visual/addons/report/VReport.kt (hunk 3 above)
Comment (hfazai) [reply galite-core/src/main/kotlin/org/kopi/galite/visual/addons/report/VReport.kt:28]: Same

**Pull request merged**